### PR TITLE
Feature/activate account ensure back links on routes are behaving correctly

### PR DESCRIPTION
**Pull request by omarCreativeDev**

### CODE REVIEW

**Comment by omarCreativeDev** [comment]
[(10) User Accounts - Activate Account](https://trello.com/c/CjBwDiN1/142-10-user-accounts-activate-account)

File: src/app/core/model/summary-list.model.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export interface SummaryList {
   label: string;
   data: string;
-  route?: string;
+  route?: string | any[];
```
Comment (ChazUK) [inline src/app/core/model/summary-list.model.ts]: It'd probably be a good idea to have it as just `any[]`, and do we need to include fragments?

File: src/app/features/activate-user-account/security-question/security-question.component.ts [hunk 2]
```diff
@@ -27,22 +26,23 @@ export class SecurityQuestionComponent extends SecurityQuestion {
...
+    this.setupSubscription();
+    this.setBackLink();
   }
 
   protected setBackLink(): void {
-    const route: string = this.securityDetailsExist ? 'confirm-account-details' : 'create-username';
-    this.backService.setBackLink({ url: ['/activate-account', this.activationToken, route] });
+    this.return = this.createAccountService.returnTo$.value;
+    this.back = this.return ? this.return : { url: ['/activate-account', this.activationToken, 'create-username'] };
+    console.log('setBackLink fired', this.return, this.back);
```
Comment (ChazUK) [inline src/app/features/activate-user-account/security-question/security-question.component.ts]: console.log

Comment (omarCreativeDev) [reply src/app/features/activate-user-account/security-question/security-question.component.ts]: removed

File: src/app/core/model/summary-list.model.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/core/model/summary-list.model.ts]: good spot, have typed as `URLStructure` for consistency

File: src/app/features/activate-user-account/security-question/security-question.component.ts (hunk 2 above)
Comment (omarCreativeDev) [reply src/app/features/activate-user-account/security-question/security-question.component.ts]: removed

**Pull request merged**